banking-app/test_docker.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+set -euo pipefail
+
+# create certificate files
+create_certificate(){
+    local certName="$1"
+    local certFile="${1}_cert.pem"
+    local setUserFile="set_${1}.json"
+    /opt/ccf/bin/keygenerator.sh --name $certName --gen-enc-key
+}
+
+
+# Prepare a test network by adding a member 
+# and two users to the network, and then open it
+create_test_network_proposal(){
+
+    create_certificate "user0"
+    create_certificate "user1"
+    create_certificate "member1"
+
+    local user0_cert=$(< user0_cert.pem sed '$!G' | paste -sd '\\n' -)
+    local user1_cert=$(< user1_cert.pem sed '$!G' | paste -sd '\\n' -)
+    local member1_cert=$(< member1_cert.pem sed '$!G' | paste -sd '\\n' -)
+    local member1_encryption_pub_key=$(< member1_enc_pubk.pem sed '$!G' | paste -sd '\\n' -)
+    local service_cert=$(< service_cert.pem sed '$!G' | paste -sd '\\n' -)
+
+    local proposalFileName="network_open_proposal.json"
+    cat <<JSON > $proposalFileName
+{
+  "actions": [
+    {
+      "name": "set_member",
+      "args": {
+        "cert": "${member1_cert}\n",
+        "encryption_pub_key": "${member1_encryption_pub_key}\n"
+      }
+    },
+    {
+      "name": "set_user",
+      "args": {
+        "cert": "${user0_cert}\n"
+      }
+    },
+    {
+      "name": "set_user",
+      "args": {
+        "cert": "${user1_cert}\n"
+      }
+    },
+    {
+      "name": "transition_service_to_open",
+      "args": {
+        "next_service_identity": "${service_cert}\n"
+      }
+    }
+  ]
+}
+JSON
+}
+
+##############################################
+# Discover docker configuration
+##############################################
+containerId=$(docker ps -f ancestor=banking-app:virtual -q)
```

Comment:
this script will only allow you to test the virtual container image. 
is that intentional?

Reply:
Great spot - I never tested the enclave version - updated